app/models/host/base.rb

```diff
@@ -127,32 +127,6 @@ def self.import_host(hostname, certname = nil)
       host
     end
 
-    def create_new_host_when_facts_are_uploaded?
-      Setting[:create_new_host_when_facts_are_uploaded]
-    end
-
-    # expect a facts hash
-    def import_facts(facts, source_proxy = nil)
-      return false if !create_new_host_when_facts_are_uploaded? && new_record?
-
-      # we are not importing facts for hosts in build state (e.g. waiting for a re-installation)
-      raise ::Foreman::Exception.new('Host is pending for Build') if build?
-      facts = facts.with_indifferent_access
-
-      facts[:domain] = facts[:domain].downcase if facts[:domain].present?
-
-      type = facts.delete(:_type)
-      facts_importer = Foreman::Plugin.fact_importer_registry.get(type).new(self, facts)
-      telemetry_observe_histogram(:importer_facts_import_duration, facts.size, type: type)
-      telemetry_duration_histogram(:importer_facts_import_duration, 1000, type: type) do
-        facts_importer.import!
-      end
-
-      save(:validate => false)
-
-      parse_facts facts, type, source_proxy
-    end
-
```

Comment:
15 lines less, 2000 to go :-)

Reply:
yeah, I realize I've not taken an easy task xD
But it seems we can do it slowly :)